### PR TITLE
feat(databend sink): service use databend-client

### PR DESCRIPTION
**Pull request by everpcpc**
* Change to use databend-client from databend rust driver: https://github.com/datafuselabs/bendsql/
* Change config `endpoint` default to DSN style.

### CODE REVIEW

**Review by jszwedko — APPROVED** [review]
Great, thanks @everpcpc ! Always nice to delete code :) 

Are there any user-facing changes with this PR?

File: website/cue/reference/components/sinks/base/databend.cue [hunk 1]
```diff
@@ -28,13 +28,8 @@ base: components: sinks: databend: configuration: {
...
 	}
 	auth: {
-		description: """
-			Configuration of the authentication strategy for HTTP requests.
-
-			HTTP authentication should be used with HTTPS only, as the authentication credentials are passed as an
-			HTTP header without any additional encryption beyond what is provided by the transport itself.
-			"""
-		required: false
+		description: "The username&password to authenticate with, will override the username&password in DSN."
```
Comment (buraizu) [inline website/cue/reference/components/sinks/base/databend.cue:31]:
```suggestion
		description: "The username and password to authenticate with, overrides the username and password in DSN."
```

File: website/cue/reference/components/sinks/base/databend.cue [hunk 2]
```diff
@@ -122,12 +117,9 @@ base: components: sinks: databend: configuration: {
 		}
 	}
 	database: {
-		description: "The database that contains the table that data is inserted into."
+		description: "The database that contains the table that data is inserted into. will override the database in DSN."
```
Comment (buraizu) [inline website/cue/reference/components/sinks/base/databend.cue:120]:
```suggestion
		description: "The database that contains the table that data is inserted into. Overrides the database in DSN."
```

**Review by buraizu — APPROVED** [review]
Approving with minor updates requested

**Comment by everpcpc** [comment]
> Are there any user-facing changes with this PR?

@jszwedko Yes, with the new client, the `endpoint` config supports both HTTP URI like `http://localhost:8000` as well as DSN like `databend://root:@localhost:8000/mydatabase?sslmode=disable&somearg=somevalue` which could provide more customization for the inner client.

**Review by jszwedko — APPROVED** [review]
Got it, thanks! My next request was going to be to add a changelog entry, but I see you already did so 😄

**Pull request merged**